routers/web/web.go

```diff
@@ -75,7 +74,23 @@ func CorsHandler() func(next http.Handler) http.Handler {
 	}
 
 	return func(next http.Handler) http.Handler {
-		return next
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.Method == http.MethodOptions {
```

Comment:
```go
if corsHandler != nil && (r.Method != http.MethodOptions || r.Header.Get("Access-Control-Request-Method") != "") {
	corsHandler(next).ServeHTTP(w, r)
} else if r.Method != http.MethodOptions {
	next.ServeHTTP(w, r)
} else {
	// it should explicitly deny OPTIONS requests if CORS handler is not executed, to avoid the next GET/POST handler being incorrectly called by the OPTIONS request
	w.WriteHeader(http.StatusMethodNotAllowed)
}
```
?

Reply:
I believe my code is more readable. It simply reads as 2 blocks:

1. For "OPTIONS" request, do this.
2. For non-OPTIONS request, do that.